src/Catalyst.Core.Lib/IO/EventLoop/BaseLoopGroupFactory.cs

```diff
@@ -0,0 +1,108 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dawn;
+using DotNetty.Transport.Channels;
+
+namespace Catalyst.Core.Lib.IO.EventLoop
+{
+    /// <summary>
+    /// The <see cref="BaseLoopGroupFactory"/> class keeps references of event loop groups that are created
+    /// and holds them from disposing
+    /// </summary>
+    /// <seealso cref="System.IDisposable" />
+    public class BaseLoopGroupFactory : IDisposable
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fix this implementation of 'IDisposable' to conform to the dispose pattern.](https://app.codacy.com/app/catalyst-network/Catalyst.Node/pullRequest?prid=4318201)